# HG changeset patch
# User Rik <rik@octave.org>
# Date 1490832146 25200
#      Wed Mar 29 17:02:26 2017 -0700
# Node ID f04f32f08590d7bbc0c28bf58d267dd0a3d4ce0f
# Parent  9cd68be3dba8da5275a428c70fe625755ff79702
Override disp rather than display for class objects (bug #50640).

* scripts/@ftp/disp.m: print lowercase "object".

* scripts/audio/@audioplayer/disp.m: Add header describing object rather than
pretending object is a structure.

* scripts/audio/@audiorecorder/disp.m: Add header describing object rather than
pretending object is a structure.

* scripts/general/inputParser.m: Switch to overloading disp, not display.

* scripts/@ftp/display.m: Deleted.

* scripts/audio/@audioplayer/display.m: Deleted.

* scripts/audio/@audiorecorder/display.m: Deleted.

* scripts/module.mk: Remove @ftp/display.m from build system.

* scripts/audio/module.mk: Remove @audioplayer/display.m,
@audiorecorder/display.m from build system.

diff --git a/scripts/@ftp/disp.m b/scripts/@ftp/disp.m
--- a/scripts/@ftp/disp.m
+++ b/scripts/@ftp/disp.m
@@ -17,15 +17,15 @@
 ## <http://www.gnu.org/licenses/>.
 
 function disp (obj)
 
   if (nargin != 1)
     print_usage ();
   endif
 
-  printf ("  FTP Object\n");
+  printf ("  FTP object\n");
   printf ("    host: %s\n", obj.host);
   printf ("    user: %s\n", obj.username);
   printf ("     dir: %s\n", __ftp_pwd__ (obj.curlhandle));
   printf ("    mode: %s\n\n", __ftp_mode__ (obj.curlhandle));
 
 endfunction
diff --git a/scripts/@ftp/display.m b/scripts/@ftp/display.m
deleted file mode 100644
--- a/scripts/@ftp/display.m
+++ /dev/null
@@ -1,32 +0,0 @@
-## Copyright (C) 2009-2017 David Bateman
-##
-## This file is part of Octave.
-##
-## Octave is free software; you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation; either version 3 of the License, or
-## (at your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-## GNU General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <http://www.gnu.org/licenses/>.
-
-function display (obj)
-
-  if (nargin != 1)
-    print_usage ();
-  endif
-
-  printf ("%s =\n\n", inputname (1));
-  disp (obj);
-
-endfunction
-
-
-## No test possible for interactive function.
-%!assert (1)
diff --git a/scripts/audio/@audioplayer/disp.m b/scripts/audio/@audioplayer/disp.m
--- a/scripts/audio/@audioplayer/disp.m
+++ b/scripts/audio/@audioplayer/disp.m
@@ -22,12 +22,16 @@
 ## @end deftypefn
 
 function disp (player)
 
   if (nargin != 1)
     print_usage ();
   endif
 
-  disp (__get_properties__ (player));
+  printf ("audioplayer object with properties:\n\n");
+  for [val, prop] = __get_properties__ (player)
+    printf ("  %s = ", prop), disp (val);
+  endfor
+  ## FIXME: display in pr-output.cc should add "\n", not this function. 
   printf ("\n");
 
 endfunction
diff --git a/scripts/audio/@audioplayer/display.m b/scripts/audio/@audioplayer/display.m
deleted file mode 100644
--- a/scripts/audio/@audioplayer/display.m
+++ /dev/null
@@ -1,33 +0,0 @@
-## Copyright (C) 2013-2017 Vytautas Jančauskas
-##
-## This file is part of Octave.
-##
-## Octave is free software; you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation; either version 3 of the License, or
-## (at your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-## GNU General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <http://www.gnu.org/licenses/>.
-
-## -*- texinfo -*-
-## @deftypefn {} {} display (@var{player})
-## Display the properties of the audioplayer object @var{player}.
-## @end deftypefn
-
-function display (player)
-
-  if (nargin != 1)
-    print_usage ();
-  endif
-
-  printf ("%s =\n", inputname (1));
-  disp (player);
-
-endfunction
diff --git a/scripts/audio/@audiorecorder/disp.m b/scripts/audio/@audiorecorder/disp.m
--- a/scripts/audio/@audiorecorder/disp.m
+++ b/scripts/audio/@audiorecorder/disp.m
@@ -22,12 +22,16 @@
 ## @end deftypefn
 
 function disp (recorder)
 
   if (nargin != 1)
     print_usage ();
   endif
 
-  disp (__get_properties__ (recorder));
+  printf ("audiorecorder object with properties:\n\n");
+  for [val, prop] = __get_properties__ (recorder)
+    printf ("  %s = ", prop), disp (val);
+  endfor
+  ## FIXME: display in pr-output.cc should add "\n", not this function. 
   printf ("\n");
 
 endfunction
diff --git a/scripts/audio/@audiorecorder/display.m b/scripts/audio/@audiorecorder/display.m
deleted file mode 100644
--- a/scripts/audio/@audiorecorder/display.m
+++ /dev/null
@@ -1,33 +0,0 @@
-## Copyright (C) 2013-2017 Vytautas Jančauskas
-##
-## This file is part of Octave.
-##
-## Octave is free software; you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation; either version 3 of the License, or
-## (at your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-## GNU General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <http://www.gnu.org/licenses/>.
-
-## -*- texinfo -*-
-## @deftypefn {} {} display (@var{recorder})
-## Display the properties of the audiorecorder object @var{recorder}.
-## @end deftypefn
-
-function display (recorder)
-
-  if (nargin != 1)
-    print_usage ();
-  endif
-
-  printf ("%s =\n", inputname (1));
-  disp (recorder);
-
-endfunction
diff --git a/scripts/audio/module.mk b/scripts/audio/module.mk
--- a/scripts/audio/module.mk
+++ b/scripts/audio/module.mk
@@ -9,33 +9,31 @@ scripts_audio_FCN_FILES = \
   scripts/audio/record.m \
   scripts/audio/sound.m \
   scripts/audio/soundsc.m
 
 scripts_audio_@audioplayer_FCN_FILES = \
   scripts/audio/@audioplayer/__get_properties__.m \
   scripts/audio/@audioplayer/audioplayer.m \
   scripts/audio/@audioplayer/disp.m \
-  scripts/audio/@audioplayer/display.m \
   scripts/audio/@audioplayer/get.m \
   scripts/audio/@audioplayer/isplaying.m \
   scripts/audio/@audioplayer/pause.m  \
   scripts/audio/@audioplayer/play.m \
   scripts/audio/@audioplayer/playblocking.m \
   scripts/audio/@audioplayer/resume.m \
   scripts/audio/@audioplayer/set.m \
   scripts/audio/@audioplayer/stop.m \
   scripts/audio/@audioplayer/subsasgn.m \
   scripts/audio/@audioplayer/subsref.m
 
 scripts_audio_@audiorecorder_FCN_FILES = \
   scripts/audio/@audiorecorder/__get_properties__.m \
   scripts/audio/@audiorecorder/audiorecorder.m \
   scripts/audio/@audiorecorder/disp.m \
-  scripts/audio/@audiorecorder/display.m \
   scripts/audio/@audiorecorder/get.m \
   scripts/audio/@audiorecorder/getaudiodata.m \
   scripts/audio/@audiorecorder/getplayer.m \
   scripts/audio/@audiorecorder/isrecording.m \
   scripts/audio/@audiorecorder/pause.m \
   scripts/audio/@audiorecorder/play.m \
   scripts/audio/@audiorecorder/record.m \
   scripts/audio/@audiorecorder/recordblocking.m \
diff --git a/scripts/general/inputParser.m b/scripts/general/inputParser.m
--- a/scripts/general/inputParser.m
+++ b/scripts/general/inputParser.m
@@ -461,27 +461,27 @@ classdef inputParser < handle
           if (vidx++ < pnargin && this.KeepUnmatched)
             this.Unmatched.(name) = varargin{vidx};
           else
             this.error (sprintf ("argument '%s' is not a valid parameter",
                                  toupper (name)));
           endif
         endif
       endwhile
-      ## Add them to the UsingDeafults list
+      ## Add them to the UsingDefaults list
       this.add_missing ("Parameter");
       this.add_missing ("Switch");
 
     endfunction
 
-    function display (this)
+    function disp (this)
       if (nargin != 1)
         print_usage ();
       endif
-      printf ("%s = inputParser object with properties:\n\n", inputname ());
+      printf ("inputParser object with properties:\n\n");
       b2s = @(x) ifelse (any (x), "true", "false");
       printf (["   CaseSensitive   : %s\n   FunctionName    : %s\n" ...
                "   KeepUnmatched   : %s\n   PartialMatching : %s\n" ...
                "   StructExpand    : %s\n\n"],
                b2s (this.CaseSensitive), b2s (this.FunctionName),
                b2s (this.KeepUnmatched), b2s (this.PartialMatching),
                b2s (this.StructExpand));
       printf ("Defined parameters:\n\n   {%s}\n",
diff --git a/scripts/module.mk b/scripts/module.mk
--- a/scripts/module.mk
+++ b/scripts/module.mk
@@ -51,17 +51,16 @@ FCN_FILE_DIRS += scripts/@ftp
 scripts_@ftp_FCN_FILES = \
   scripts/@ftp/ascii.m \
   scripts/@ftp/binary.m  \
   scripts/@ftp/cd.m  \
   scripts/@ftp/close.m  \
   scripts/@ftp/delete.m  \
   scripts/@ftp/dir.m  \
   scripts/@ftp/disp.m  \
-  scripts/@ftp/display.m  \
   scripts/@ftp/ftp.m  \
   scripts/@ftp/loadobj.m  \
   scripts/@ftp/mget.m  \
   scripts/@ftp/mkdir.m  \
   scripts/@ftp/mput.m  \
   scripts/@ftp/rename.m  \
   scripts/@ftp/rmdir.m  \
   scripts/@ftp/saveobj.m
